packages/provider/src/MultiChainProvider.test.ts

```diff
@@ -0,0 +1,404 @@
+import { SnapProvider } from '@metamask/snap-types';
+import {
+  getSessionNamespace,
+  getRequestNamespace,
+} from '@metamask/snap-utils/test-utils';
+import { NamespaceId, RequestNamespace } from '@metamask/snap-utils';
+import { MultiChainProvider } from './MultiChainProvider';
+
+Object.assign(globalThis, {
+  ethereum: {
+    on: jest.fn(),
+    request: jest.fn(),
+  },
+});
+
+// Used for mocking the provider's return values.
+declare const ethereum: SnapProvider;
+
+/**
+ * Get a new `MultiChainProvider` instance, that is connected to the given
+ * namespaces.
+ *
+ * @param namespaces - The namespaces to connect to.
+ * @returns The new `MultiChainProvider` instance.
+ */
+async function getProvider(
+  namespaces: Record<NamespaceId, RequestNamespace> = {
+    eip155: getRequestNamespace(),
+  },
+) {
+  const request = ethereum.request as jest.MockedFunction<
+    typeof ethereum.request
+  >;
+
+  request.mockImplementation(async () => ({
+    jsonrpc: '2.0',
+    id: 1,
+    result: {
+      namespaces: {
+        eip155: getSessionNamespace(),
+      },
+    },
+  }));
+
+  const provider = new MultiChainProvider();
+
+  const { approval } = await provider.connect({
+    requiredNamespaces: namespaces,
+  });
+  await approval();
+
+  return provider;
+}
+
+describe('MultiChainProvider', () => {
+  describe('connect', () => {
```

Comment:
If we're using mocks in `globalThis.ethereum` we should reset them.

```typescript
beforeEach(() => {
  jest.resetAllMocks();
});

Reply:
This is done automatically by Jest.

https://github.com/MetaMask/snaps-skunkworks/blob/3be288697003f6a4263cfc8c948e210a90a6b401/packages/provider/jest.config.js#L18-L24